toolchain/check/import_ref.cpp

```diff
@@ -733,8 +730,10 @@ class ImportRefResolver {
          .param_refs_id =
              GetLocalParamRefsId(function.param_refs_id, param_const_ids),
          .return_type_id = new_return_type_id,
-         .return_slot_id = new_return_slot,
+         .return_storage_id = new_return_storage,
          .is_extern = function.is_extern,
+         // There has been no call or definition in this file.
+         .return_slot = SemIR::Function::ReturnSlot::NotComputed,
```

Comment:
Should this be forwarded when Absent or similar? If it's an Error in an import, should it become valid when imported? i.e., if known from the API, I'm not confident it should be ignored.

As an example test:

```
library "a" api;
class C;
// Diagnoses incomplete type, right?
fn F() -> C;
```

```
library "b" api;
import library "a";

fn G() {
  // Will this diagnose incomplete?
  F();
}
```

```
library "c" api;
import library "a";

class C {}

fn H() {
  // Does this get a different result for the return slot?
  F();
}
```

Note, that'd also have implications for merging, since it's not part of the prev_function merge right now.

Reply:
When the return type is incomplete, the `Error` state gets re-checked on every call anyway; the difference between `NotChecked` and `Error` is that `Error` suppresses diagnostics. So I think it's just a question of whether we want a diagnostic per file, or one in total.

Per in-person discussion, more error suppression is probably better, so changed this to import and merge the return slot information.